src/FakeItEasy/Core/DefaultFakeObjectCallFormatter.cs

```diff
@@ -164,36 +164,40 @@ private void AppendArgumentsList(StringBuilder builder, IFakeObjectCall call)
             }
         }
 
-        private void AppendArgumentValue(StringBuilder builder, ArgumentValueInfo argument)
+        private void AppendArgumentValue(StringBuilder builder, int argumentIndex, ArgumentValueInfo argument)
         {
+            // Usually parameters will be named, but in F# (at least) it's possible
+            // to declare a method with anonymous parameters. In that case, we try to
+            // help the user by outputting names like "param1", "param2", …
+            string argumentName = argument.ArgumentName ?? $"param{argumentIndex + 1}";
             builder
-                .Append(argument.ArgumentName)
+                .Append(argumentName)
                 .Append(": ")
                 .Append(this.GetArgumentValueAsString(argument.ArgumentValue));
         }
 
-        private string GetArgumentValueAsString(object argumentValue)
+        private string GetArgumentValueAsString(object? argumentValue)
         {
             return this.argumentValueFormatter.GetArgumentValueAsString(argumentValue);
         }
 
         private void AppendArguments(StringBuilder builder, ArgumentValueInfo[] arguments)
         {
             var totalNumberOfArguments = arguments.Length;
-            var callIndex = 0;
+            var argumentIndex = 0;
             foreach (var argument in arguments)
             {
-                AppendArgumentSeparator(builder, callIndex, totalNumberOfArguments);
-                this.AppendArgumentValue(builder, argument);
-                callIndex++;
+                AppendArgumentSeparator(builder, argumentIndex, totalNumberOfArguments);
+                this.AppendArgumentValue(builder, argumentIndex, argument);
+                argumentIndex++;

```

Comment:
nit: the argument index could be in `ArgumentValueInfo`, making this code slightly simpler

Reply:
Okay! I tried a thing. Tell me what you think.